src-electron/db/query-upgrade.js

```diff
@@ -0,0 +1,118 @@
+/**
+ *
+ *    Copyright (c) 2020 Silicon Labs
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+const dbApi = require('./db-api.js')
+const dbEnum = require('../../src-shared/db-enum.js')
+const fs = require('fs')
+const fsp = fs.promises
+
+async function selectClusterName(db, clusterRef) {
+  let clusterName = await dbApi.dbAll(
+    db,
+    'SELECT NAME FROM CLUSTER WHERE CLUSTER_ID = ?',
+    [clusterRef]
+  )
+  return clusterName[0].NAME
+}
+
+async function selectAttributeName(db, attributeId) {
+  let attributeName = await dbApi.dbAll(
+    db,
+    'SELECT NAME FROM ATTRIBUTE WHERE ATTRIBUTE_ID = ?',
+    [attributeId]
+  )
+  return attributeName[0].NAME
+}
+
+async function queryMetaFile(db, packageId) {
+  let path = await dbApi.dbAll(
+    db,
+    'SELECT PATH FROM PACKAGE WHERE PACKAGE_ID = ?',
+    [packageId]
+  )
+  return path[0].PATH
+}
+async function queryPackages(db, attributeId) {
+  let package_ref = await dbApi.dbAll(
+    db,
+    'SELECT PACKAGE_REF FROM ATTRIBUTE WHERE ATTRIBUTE_ID = ?',
+    [attributeId]
+  )
+  return package_ref[0].PACKAGE_REF
+}
+
+/**
+ * Returns an array of objects containing global attributes that should be forced external.
+ *
+ * @export
+ * @param {*} db
+ * @param {*} attributeId
+ * @returns An array of objects
+ */
+
+async function checkGlobals(db, attributeId) {
+  let pkgs = await queryPackages(db, attributeId)
```

Comment:
This seems off to me. Don't you have packagesIds in importAttributeFromEndpointType?

Reply:
huh?

Reply:
its getting the path to the json file